Mathlib/Algebra/Order/Monoid/Canonical/Defs.lean

```diff
@@ -17,12 +17,44 @@ universe u
 
 variable {α : Type u}
 
+/-- An ordered additive monoid is `CanonicallyOrderedAdd`
+  if the ordering coincides with the subtractibility relation,
+  which is to say, `a ≤ b` iff there exists `c` with `b = a + c`.
+  This is satisfied by the natural numbers, for example, but not
+  the integers or other nontrivial `OrderedAddCommGroup`s. -/
+class CanonicallyOrderedAdd (α : Type*) [Add α] [LE α] extends
+  ExistsAddOfLE α : Prop where
+  /-- For any `a` and `b`, `a ≤ a + b` -/
+  protected le_self_add : ∀ a b : α, a ≤ a + b
+
+attribute [instance 50] CanonicallyOrderedAdd.toExistsAddOfLE
+
+/-- An ordered monoid is `CanonicallyOrderedMul`
+  if the ordering coincides with the divisibility relation,
+  which is to say, `a ≤ b` iff there exists `c` with `b = a * c`.
+  Examples seem rare; it seems more likely that the `OrderDual`
+  of a naturally-occurring lattice satisfies this than the lattice
+  itself (for example, dual of the lattice of ideals of a PID or
+  Dedekind domain satisfy this; collections of all things ≤ 1 seem to
+  be more natural that collections of all things ≥ 1).
```

Comment:
We might want to flip this around! Let's not do that in this PR though.

But we also see that with valuations: additive ones naturally take values in a `WithTop A`, where `A` is a linearly ordered additive comm group. But multiplicative ones naturally take values in `WithZero G`. The `WithTop` vs `WithZero` is an artifact of flipping the order when moving between the two notions.